.github/workflows/ci.yml

```diff
@@ -134,7 +134,12 @@ jobs:
         with:
           go-version: ${{ needs.checks.outputs.go_version }}
       - name: Run Tests
-        run: go test ./...
+        run: make cover
+      - name: Upload coverage to Codecov
```

Comment:
why are we uploading our coverage to Codecov?

Reply:
to get the report (the first comment in the PR) and I guess to have a history to compare to for the coverage diff

Reply:
Sorry, I guess I should be more clear. Why do we upload our code coverage data to a 3rd party website ?
Also, the description of you PR is very minimal, it states what you have done, not why.

Reply:
I'm not sure what more there is to explain. Codecov is a tool that shows the coverage for the PR and how it changes from the baseline. In order to have this information in the PR (and for the codebase), you need to upload the data to their website to be analyzed. You can find out more about it here https://about.codecov.io/for/open-source/

What is the concern with uploading the coverage result to Codecov?

By the way, this process is similar for other tools we use like Fossa and Code Scanning in GitHub.